Cannot build with opengl3=false on macos

### Tested versions

- Reproductible on master (06827c91c6ee98068a48c66be0cc067ebb434a2f) and on 4.5-stable tag (876b290332ec6f2e6d173d08162a02aa7e6ca46d)

### System information

macos arm (M3)

### Issue description

Can't compile engine with opengl disabled on macos. Works alright in windows.

```
In file included from platform/macos/embedded_gl_manager.mm:34:
In file included from platform/macos/platform_gl.h:47:
./thirdparty/glad/glad/egl.h:326:10: fatal error: 'KHR/khrplatform.h' file not found
  326 | #include <KHR/khrplatform.h>
      |          ^~~~~~~~~~~~~~~~~~~
1 error generated.
scons: *** [bin/obj/platform/macos/embedded_gl_manager.macos.editor.arm64.o] Error 1
scons: building terminated because of errors.
```

### Steps to reproduce

On a clean clone on macos arm, try to compile engine with:
`scons platform=macos arch=arm64 opengl3=false`

### Minimal reproduction project (MRP)

Not necessary, just need engine

--- a/platform/macos/embedded_gl_manager.mm
+++ b/platform/macos/embedded_gl_manager.mm
@@ -30,10 +30,10 @@
 
 #import "embedded_gl_manager.h"
 
+#if defined(MACOS_ENABLED) && defined(GLES3_ENABLED)
+
 #import "drivers/gles3/storage/texture_storage.h"
 #import "platform_gl.h"
-
-#if defined(MACOS_ENABLED) && defined(GLES3_ENABLED)
 
 #import <QuartzCore/QuartzCore.h>
 #include <dlfcn.h>

